app/Http/Controllers/PublicUrl.php

```diff
@@ -0,0 +1,14 @@
+<?php
+
+namespace App\Http\Controllers;
+
+/**
+ * @codeCoverageIgnore
+ */
+trait PublicUrl
+{
+    private static function _publicUrl($file, $disk = self::DISK)
```

Comment:
The whole code works, my only concern here is that nothing enforces the class using this trait to have a `DISK` property. Do you have any better idea how to enforce the consuming class to have the property this needs? If not feel free to merge this PR as is.

Reply:
Well, defining a property in the trait would be the best, if we could override it - sadly, it is not simply possible...
https://stackoverflow.com/questions/32571920/overriding-doctrine-trait-properties
I think I might be able to move the trait usage into the controller base class, which I subclass in every other controller...

Reply:
How about that instead?
A default for a default...

Reply:
Sure, then feel free to merge this